File: Arrays: Carry Forward/agPairs.py
class Solution:
    # @param A : string
    # @return an integer
    def solve(self, A):
        n = len(A)
        c = 0
        ans = 0

        for i in range(n - 1, -1, -1):
            if A[i] == "g":
                c += 1
            else:
                ans = ans + c

        return ans

File: Arrays: Carry Forward/AgPairs.java
public class AgPairs {
    public int solve(String A) {
        int n = A.length(), ans = 0, MOD = 1000 * 1000 * 1000 + 7;
        int c = 0;

        for (int i = n - 1; i >= 0; i--) {
            if (A.charAt(i) == 'G')
                c++;
            else if (A.charAt(i) == 'A')
                ans = (ans + c) % MOD;
        }

        return ans;
    }
}